caproto/curio/client.py

```diff
@@ -35,10 +35,11 @@ def __init__(self, circuit):
         self.new_command_condition = curio.Condition()
         self._socket_lock = curio.Lock()
 
-    async def create_connection(self):
-        self.socket = await socket.create_connection(self.circuit.address)
-        await self._socket_lock.release()
+    async def connect(self):
```

Comment:
The original version of this is pretty clever but possibly buggy. Its advantage: returning a Channel object before the socket was finished opening. I retreated to a blunter approach for simplicity, but ultimate we might want to restore the original one before merging.

Reply:
If I recall correctly, I think it's because I ran into the case where something else had locked the socket and `connect()` was unable to do its thing.

Reply:
I think we should merge this change, actually. The existing code can error out if the `socket` isn't ready by the time the `_receive_loop` task is spawned.The original goal of the complexity here was to make channel creation as asynchronous as possible. We can revisit that goal when we do performance tuning.